From 7a5ceb808a91263bb5f652c583ccb78f39d6ddb9 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Fri, 29 Mar 2013 16:45:59 -0700
Subject: [PATCH 153/382] MIPS: Octeon: Add Octeon3 cache handling

Based on SDK octeon3_3.10.

- added cache size information for OIII
- fixed clearing of pending errors from COP0_CACHE_ERR, its not present in OIII

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
Signed-off-by: Leonid Rosenboim <lrosenboim@caviumnetworks.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/mm/c-octeon.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/mips/mm/c-octeon.c b/arch/mips/mm/c-octeon.c
index 86c9596..fb351e8 100644
--- a/arch/mips/mm/c-octeon.c
+++ b/arch/mips/mm/c-octeon.c
@@ -314,6 +314,9 @@ static void __cpuinit probe_octeon(void)
 		c->dcache.sets = 8;
 		dcache_size = c->dcache.sets * c->dcache.ways * c->dcache.linesz;
 		c->options |= MIPS_CPU_PREFETCH;
+
+		board_be_handler = octeon2_be_handler;
+		board_mcheck_handler = octeon2_mcheck_handler;
 		break;
 
 	default:
-- 
1.7.0.4

